### PR TITLE
feat(docs): Release adoption stages filtering

### PR DESCRIPTION
**Pull request by taylangocmen**
This adds search features for the `release.stage` feature to the #3903

### CODE REVIEW

File: src/docs/product/releases/usage/sorting-filtering.mdx [hunk 1]
```diff
@@ -22,11 +22,30 @@ On the **Releases** page, you can use the "Sort By" dropdown to sort releases by
...
+If you are using our [Semantic Versioning](/platform-redirect/?next=/configuration/releases/%23bind-the-version) format, you can use the following query tokens:
 
 - `release` - Search based on string comparison
 - `release.package` - Search releases with matching package names
 - `release.version` - Search releases with matching semantic version. Supports ranges and wildcards (for example, `release.version:>7.8` or `release.version:1.2.*`)
 - `release.build` - Search releases with matching build numbers
 
+#### Release Stages
+
+<Note>
```
Comment (adhiraj) [inline src/docs/product/releases/usage/sorting-filtering.mdx:38]: We have the same note at the page-level, so is this redundant? Or that other note is meant to be for Sorting? ... in which case we should move it into the Sorting section

File: src/docs/product/releases/usage/sorting-filtering.mdx [hunk 2]
```diff
@@ -22,11 +22,30 @@ On the **Releases** page, you can use the "Sort By" dropdown to sort releases by
...
+  Release Stages are available only if you're in the Early Adopter program. Features available to Early Adopters are still in-progress and may have bugs. We recognize the irony. If you’re interested in being an Early Adopter, you can turn your organization’s Early Adopter status on/off in General Settings. This will affect all users in your organization and can be turned back off just as easily.
+
+</Note>
+
+If you select a single environment on the releases page, each project will be labeled with one of the following:
+- Adopted (`adopted`): when a release is adopted, meaning it was seen in a high percentage of sessions for the project
+- Low adoption (`low_adoption`): when a release has low adoption, meaning it was seen in a low percentage of sessions for the project
+- Replaced (`replaced`): when a release was replaced by a release before it was adopted, meaning later releases were seen at a higher percentage of sessions before this release
+
+You can use the following query with one of the release  labels, ie. `adopted`, `low_adoption`, `replaced`:
```
Comment (adhiraj) [inline src/docs/product/releases/usage/sorting-filtering.mdx:49]: The query is missing?

File: src/docs/product/releases/health/index.mdx [hunk 3]
```diff
@@ -83,23 +83,23 @@ Each release is represented by a line in the chart, and you can navigate to the
...
-- "Adopted" - the release's percentage of sessions across all sessions is above a threshold (which is 10% divided by the number of releases in the last six hours)
-- "Low Adoption" - the release's percentage of sessions across all session that are below the threshold
-- "Replaced" -  the release was previously tagged as "Adopted", but no longer meets the minimum threshold. "Replaced" releases that reach the threshold will once again be marked as "High Adoption".
-
+Adoption stage is a high-level overview of the usage your release is seeing relative to other releases. It is calculated every hour by processing your sessions over the last six hours.
+For each release, the release's percentage of sessions against all sessions in the same environment and project is calculated and marked as being in one of three stages based on a threshold of 10%. These stages are:
```
Comment (imatwawana) [inline src/docs/product/releases/health/index.mdx:95]:
```suggestion
For each release, the release's percentage of sessions is calculated against all sessions in the same environment and project. It's then marked as being in one of three stages, based on a threshold of 10%. These stages are:
```

File: src/docs/product/releases/health/index.mdx [hunk 4]
```diff
@@ -83,23 +83,23 @@ Each release is represented by a line in the chart, and you can navigate to the
...
+- "Adopted" - the release's percentage of sessions across all sessions is above the threshold, ie. the release was seen in 10% or more of sessions over all the releases in the last six hours.
```
Comment (imatwawana) [inline src/docs/product/releases/health/index.mdx:96]:
```suggestion
- Adopted - The release's percentage of sessions compared to all sessions is above the threshold. That is, the release was seen in 10% or more of sessions over all releases in the last six hours.
```

File: src/docs/product/releases/health/index.mdx [hunk 5]
```diff
@@ -83,23 +83,23 @@ Each release is represented by a line in the chart, and you can navigate to the
...
+- "Low Adoption" - the release's percentage of sessions across all session that are below the threshold, ie. the release was seen in less than 10% of sessions over all the releases in the last six hours.
```
Comment (imatwawana) [inline src/docs/product/releases/health/index.mdx:97]:
```suggestion
- Low Adoption - The release's percentage of sessions compared to all sessions that are below the threshold. That is, the release was seen in less than 10% of sessions over all the releases in the last six hours.
```

File: src/docs/product/releases/health/index.mdx [hunk 6]
```diff
@@ -83,23 +83,23 @@ Each release is represented by a line in the chart, and you can navigate to the
...
+- "Replaced" -  the release was previously tagged as "Adopted", but no longer meets the minimum threshold. "Replaced" releases that reach the threshold will once again be marked as "Adopted".
 
 <Note>
 
-Adoption Stage elements in sentry.io are displayed only when filtered to a single environment.
+Adoption Stage labels are only displayed when filtered down to a single environment.
```
Comment (imatwawana) [inline src/docs/product/releases/health/index.mdx:102]:
```suggestion
Adoption Stage labels are only displayed when you filter down to a single environment.
```

File: src/docs/product/releases/health/index.mdx [hunk 7]
```diff
@@ -83,23 +83,23 @@ Each release is represented by a line in the chart, and you can navigate to the
...
-- "Replaced" -  the release was previously tagged as "Adopted", but no longer meets the minimum threshold. "Replaced" releases that reach the threshold will once again be marked as "High Adoption".
-
+Adoption stage is a high-level overview of the usage your release is seeing relative to other releases. It is calculated every hour by processing your sessions over the last six hours.
+For each release, the release's percentage of sessions against all sessions in the same environment and project is calculated and marked as being in one of three stages based on a threshold of 10%. These stages are:
+- "Adopted" - the release's percentage of sessions across all sessions is above the threshold, ie. the release was seen in 10% or more of sessions over all the releases in the last six hours.
+- "Low Adoption" - the release's percentage of sessions across all session that are below the threshold, ie. the release was seen in less than 10% of sessions over all the releases in the last six hours.
+- "Replaced" -  the release was previously tagged as "Adopted", but no longer meets the minimum threshold. "Replaced" releases that reach the threshold will once again be marked as "Adopted".
```
Comment (imatwawana) [inline src/docs/product/releases/health/index.mdx:98]:
```suggestion
- Replaced -  The release was previously tagged as "Adopted", but no longer meets the minimum threshold. "Replaced" releases that reach the threshold will once again be marked as "Adopted".
```

File: src/docs/product/releases/usage/sorting-filtering.mdx [hunk 8]
```diff
@@ -28,5 +28,6 @@ Search on the **Releases** page, which supports both raw text and query syntax.
 - `release.package` - Search releases with matching package names
 - `release.version` - Search releases with matching semantic version. Supports ranges and wildcards (for example, `release.version:>7.8` or `release.version:1.2.*`)
 - `release.build` - Search releases with matching build numbers
+- `release.stage` - Search releases with matching adoption stage, can be one of `adopted`, `low_adoption` or `replaced`. Learn more about [Release Adoption Stages](product/releases/health/#adoption-stages).
```
Comment (imatwawana) [inline src/docs/product/releases/usage/sorting-filtering.mdx:31]:
```suggestion
- `release.stage` - Search releases with matching adoption stage. Can be `adopted`, `low_adoption` or `replaced`. Learn more about [Release Adoption Stages](/product/releases/health/#adoption-stages).
```

File: src/docs/product/sentry-basics/search/searchable-properties.mdx [hunk 9]
```diff
@@ -50,6 +50,7 @@ Below is a list of event-level keys and tokens reserved and known to Sentry:
 | environment | Restrict results to events tagged with the matching environment. |
 | release | Restrict results to events tagged with the matching release. You can create a token with an exact match on the version of a release, or `release:latest` to pick the most recent release. |
 | release.package<br /> release.version<br /> release.build | Restrict results to events tagged with the matching semantic version. Supports ranges and wildcards. Learn more in [Sorting & Filtering Releases](/product/releases/usage/sorting-filtering/). |
+| release.stage | Restrict results to releases that are tagged with the matching release stage. See [Filtering Releases](product/releases/usage/sorting-filtering/#filtering-releases) for supported values. |
```
Comment (imatwawana) [inline src/docs/product/sentry-basics/search/searchable-properties.mdx:53]:
```suggestion
| release.stage | Restrict results to releases that are tagged with the matching release stage. See [Filtering Releases](/product/releases/usage/sorting-filtering/#filtering-releases) for supported values. |
```

File: src/docs/product/releases/usage/sorting-filtering.mdx (hunk 8 above)
Comment (imatwawana) [inline src/docs/product/releases/usage/sorting-filtering.mdx:31]:
```suggestion
- `release.stage` - Search releases with matching adoption stage, can be one of `adopted`, `low_adoption` or `replaced`. Learn more about [Release Adoption Stages](/product/releases/health/#adoption-stages).
```

**Review by imatwawana — APPROVED** [review]
I fixed a few links that were missing a forward slash--please check those are working before you merge. Otherwise, LGTM!

**Pull request merged**